Libs/MRML/Core/vtkMRMLAbstractViewNode.cxx

```diff
@@ -599,3 +602,21 @@ void vtkMRMLAbstractViewNode::SetAxisLabel(int labelIndex, const char* label)
   this->AxisLabels->SetValue(labelIndex, label);
   this->Modified();
 }
+
+//----------------------------------------------------------------------------
+vtkMRMLNode* vtkMRMLAbstractViewNode::GetParentLayoutNode()
+{
+  return this->GetNodeReference(this->ParentLayoutNodeReferenceRole);
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLAbstractViewNode::SetAndObserveParentLayoutNode(vtkMRMLNode* node)
```

Comment:
Pointer-based reference setting is a convenience function, but we always need to have ID-based setting (vtkMRMLAbstractViewNode::SetAndObserveParentLayoutNodeID) as well, because pointer-based setting does not work in all cases (for example, when we need to set node reference before adding the node to the scene).